### PR TITLE
Fix stake burning issue

### LINKED ISSUE #401: Bugfix: Return stake
Before: Lose 100% if proposal fails for any reason

Desired / After:
Lose 5% if it doesn't reach quorum
Lose 100% if it doesn't reach unity

## Spec
If unity is not met, burn the stake (same as before)
if unity is met, but quorum is not met, burn part of the stake, return the rest to the sponsor (whoever paid for it to begin with)
if both unity and quorum are met, the whole stake is returned (as before)

### PR DESCRIPTION
**Pull request by tlacloc**
Bugfix: Return stake
 Fix #401 
 
Now when a proposal is rejected (passing unity but with invalid quorum) user gets 95% refund

Changes in 

- [ ] seeds.proposals
- [ ] seeds.dao

### CODE REVIEW

File: src/seeds.proposals.cpp [hunk 1]
```diff
@@ -528,8 +528,13 @@ void proposals::evalproposal (uint64_t proposal_id, uint64_t prop_cycle) {
       }
 
     } else {
+
+      asset stackToBurn = valid_quorum ?  pitr->staked : asset(pitr->staked.amount * 0.05, seeds_symbol);
```
Comment (n13) [inline src/seeds.proposals.cpp:532]: typo - should be "stakeToBurn" - please fix

Comment (n13) [reply src/seeds.proposals.cpp:532]: Code looks great.

File: src/seeds.proposals.cpp [hunk 2]
```diff
@@ -528,8 +528,13 @@ void proposals::evalproposal (uint64_t proposal_id, uint64_t prop_cycle) {
...
+      stackToBurn = passed ? stackToBurn : pitr->staked;
```
Comment (n13) [inline src/seeds.proposals.cpp:533]: This is a little weird - an explicit if statement that defines all possible cases would be easier to understand. 

But - it's correct code.

File: src/seeds.proposals.cpp [hunk 3]
```diff
@@ -547,7 +552,7 @@ void proposals::evalproposal (uint64_t proposal_id, uint64_t prop_cycle) {
             proposal.passed_cycle = prop_cycle;
           }
           proposal.executed = false;
-          proposal.staked = asset(0, seeds_symbol);
+          proposal.staked = asset(pitr->staked.amount - stackToBurn.amount, seeds_symbol);
           proposal.status = status_rejected;
```
Comment (n13) [inline src/seeds.proposals.cpp:556]: Ok but it should also send back the stake now. Or does this statement alone allow the user to claim back their stake?

**Review by n13 — CHANGES_REQUESTED** [review]
Only question is this supposed to automatically send back the partial stake when unity is met, but quorum is not met?

File: src/seeds.proposals.cpp (hunk 3 above)
Comment (n13) [reply src/seeds.proposals.cpp:556]: I updated the ticket to make this much clearer. Stake should be returned.

File: src/seeds.proposals.cpp (hunk 1 above)
Comment (tlacloc) [reply src/seeds.proposals.cpp:532]: Oh, my bad, fixing it!

File: src/seeds.proposals.cpp (hunk 3 above)
Comment (tlacloc) [reply src/seeds.proposals.cpp:565]: Thanks i wasn't that sure if it was suppose to store de value of the remained or should be returned, I fix this in the new commit too

**Review by n13 — APPROVED** [review]

**Pull request merged**